# Lista de Exercícios 3

## 1. Descreva a diferença entre os testes de hipóteses paramétricos e não paramétricos. Porque optamos primeiro por usar testes paramétricos?

#### Testes Paramétricos: Esses testes são utilizados geralmente em dados que possuem o distribuição normal, são exemplos de testes paramétricos Teste t de Student, ANOVA e teste Z.
#### Testes Não Paramétricos: São livres de distribuição, Podem ser aplicados a dados nominais, ordinais, são exemplos de testes não paramétricos Teste de Mann-Whitney, teste de Wilcoxon, teste de Kruskal-Wallis e teste do Qui-Quadrado. 
#### Testes Paramétricos tem maior poder estatístico, isso implíca que havendo um efeito real na população, esses testes tem maior probabilidade de detectá-lo como estatísticamente significativo.

## 2. Teste de Hipótese para Duração da Bateria Um fabricante de telefones deseja competir no mercado de telefones touchscreen. Ele sabe que o produto líder tem bateria que dura apenas 10 horas. O fabricante afirma que, embora o novo telefone touchscreen seja mais caro, sua bateria dura mais do que a do produto líder (mais de 10 horas). Para testar a afirmação desse fabricante, um pesquisador obtém amostras de 45 unidades do novo telefone e descobre que a vida útil da bateria da amostra é, em média, 10,5 horas com um desvio padrão de 1,4 horas.


### a. Especifique as hipóteses nula e alternativa apropriadas
#### Hipótese Nula (H₀): A duração média da bateria do novo telefone é igual ou menor a 10 horas. (μ ≤ 10)
#### Hipótese Alternativa (H₁): A duração média da bateria do novo telefone é maior que 10 horas. (μ > 10)

### b. Calcule a Estatística de teste e o p-valor. Faça as validações 
necessárias para utilizar o teste

In [4]:
import numpy as np
from scipy import stats

amostra = [12.53, 12.23, 12.06, 10.54, 12.1, 10.8, 8.21, 8.79, 10.66, 10.98,
           10.07, 10.51, 10.52, 9.92, 10.87, 8.17, 10.25, 10.84, 11.75, 11.36,
           9.67, 12.19, 10.35, 11.9, 9.14, 9.84, 9.4, 12.65, 8.93, 12.41,
           8.65, 11.14, 12.01, 7.9, 10.64, 10.02, 8.07, 11.21, 10.45, 10.95,
           11.83, 10.17, 12.26, 9.46, 7.43]

n = len(amostra)
media_amostral = np.mean(amostra)
desvio_padrao_amostral = np.std(amostra, ddof=1) # ddof=1 para desvio padrão amostral
media_populacional_hipotetica = 10

t_statistic = (media_amostral - media_populacional_hipotetica) / (desvio_padrao_amostral / np.sqrt(n))

p_valor = 1 - stats.t.cdf(t_statistic, df=n-1) 

print(f"Estatística t: {t_statistic:.4f}")
print(f"p-valor: {p_valor:.4f}")

def conclusion(alpha):
    if p_value < alpha:
        return "Rejeitamos H0: A bateria do novo telefone dura mais de 10 horas."
    else:
        return "Não rejeitamos H0: Não há evidências suficientes para concluir que a bateria dura mais de 10 horas."

alpha = 0.05
print(conclusion(alpha))

Estatística t: 2.3530
p-valor: 0.0116
Rejeitamos H0: A bateria do novo telefone dura mais de 10 horas.


### c. Com um nível de significância de 5%, qual é a conclusão do teste?
#### Como o p-valor (0.0116) é menor que o nível de significância (0,05), rejeitamos a hipótese nula.).

## 3. Um estudo recente da Allstate Insurance Co. descobriu que 82% dosadolescentes americanos usam celulares enquanto dirigem. Em 2010, o estado de Massachusetts promulgou uma lei que proíbe o uso de telefone 
celular por motoristas menores de 18 anos. Um analista gostaria d 
determinar se a lei diminuiu a proporção de motoristas com menos de  8
anos que usam celular enquanto dirigem. 

### a. Especifique as hipóteses nula e alternativa.
#### Hipótese Nula (H₀): A proporção de adolescentes que usam celular enquanto dirigem em Massachusetts é igual à proporção nacional (82% ou 0,82). (p = 0,82)
#### Hipótese Alternativa (H₁): A proporção de adolescentes que usam celular enquanto dirigem em Massachusetts é menor que a proporção nacional. (p < 0,82)

### b. Suponha que uma amostra de 200 motoristas com menos de 18 anos 
resulte em 150 que ainda usam celular enquanto dirigem. Calcule  
Estatística de teste e o p-valor. Faça as validações necessárias pa a
utilizar o te.ste

In [11]:
n = 200  
x = 150  
p_a = x / n  
p0 = 0.82  
alpha = 0.05  

z_statistic = (p_a - p0) / np.sqrt((p0 * (1 - p0)) / n)

p_value = stats.norm.cdf(z_statistic)

print(f"Estatística de teste (z): {z_statistic:.4f}")
print(f"p-valor: {p_value:.5f}")
print("Rejeitamos H0" if p_value < alpha else "Não rejeitamos H0")

Estatística de teste (z): -2.5767
p-valor: 0.00499
Rejeitamos H0


### c. Com α = 0,05, a lei foi eficaz?
#### Como o p-valor (0,0049) é menor que o nível de significância (0,05), rejeitamos a hipótese nula.

## 4. A vida média de uma amostra de 100 lâmpadas de certa marca é de 1.598 horas aproximadamente. Por similaridade com outros processos de fabricação, sabe-se que o desvio padrão é igual a 120 Horas. Desejamos testar se a duração média de todas as lâmpadas dessa marca é diferente de 1.600 horas.

### a. Especifique as hipóteses nula e alternativa apropriadas. Faça as validações necessárias para utilizar o teste.
#### Hipótese Nula (H₀): A duração média das lâmpadas é igual a 1600 horas (μ = 1600).
#### Hipótese Alternativa (H₁): A duração média das lâmpadas é diferente de 1600 horas (μ ≠ 1600).

#### b. Calcule a Estatística de teste e o p-valor

In [15]:
amostra = [1647.59, 1687.1, 1673.78, 1473.45, 1416.26, 1650.44, 1584.19,
1598.89, 1563.98, 1519.6, 1424.55, 1674.44, 1619.96, 1560.52, 1486.71, 1545.91,
1910.67, 1741.09, 1532.24, 1593.26, 1663.64, 1576.65, 1445.8, 1534.2, 1568.34,
1524.93, 1578.27, 1760.9, 1572.98, 1629.74, 1773.92, 1580.99, 1569.87, 1469.23,
1487.92, 1720.54, 1631.08, 1460.43, 1555.71, 1723.44, 1568.48, 1543.65, 1543.78,
1609.93, 1599.63, 1709.23, 1434.94, 1669.04, 1388.75, 1501.29, 1623.75, 1542.41,
1617.93, 1861.44, 1557.13, 1432.54, 1766.31, 1678.37, 1463.46, 1647.16, 1611.37,
1537.3, 1617.04, 1630.58, 1618.38, 1575.04, 1774.63, 1739.11, 1693.14, 1714.35,
1698.93, 1572, 1532.64, 1418.22, 1567.97, 1470.45, 1563.12, 1639.74, 1766.52,
1672.85, 1534.74, 1375.84, 1601.11, 1650.86, 1749.66, 1630.23, 1546.1, 1697.13,
1600.64, 1646.69, 1654.1, 1666.44, 1633.33, 1490.66, 1551.56, 1434.63, 1472.35,
1734.77, 1586.06, 1604.09]

media_amostral = np.mean(amostra)
desvio_padrao_populacional = 120
tamanho_amostra = len(amostra)
media_populacional_hipotetica = 1600

z = (media_amostral - media_populacional_hipotetica) / (desvio_padrao_populacional / np.sqrt(tamanho_amostra))

p_valor = 2 * (1 - stats.norm.cdf(abs(z))) 

print(f"Estatística z: {z:.4f}")
print(f"p-valor: {p_valor:.4f}")

Estatística z: -0.1943
p-valor: 0.8459


### c. Com um nível de significância de 5%, qual é a conclusão do teste?
#### Como o p-valor (0,8459) é muito maior que o nível de significância (0,05), não rejeitamos a hipótese nula.

## 5. O mundo industrial compartilha um objetivo comum: melhorar a qualidade pela redução da variação. Os engenheiros de controle da qualidade desejam garantir que um produto tem uma média aceitável, mas desejam também, produzir itens de qualidade consistente, de modo que existirão menos defeitos. Uma empresa tem produzido latas de refrigerante com quantidades que têm desvio padrão de 0,051. Uma nova máquina de engarrafamento que foi construída com o objetivo de melhorar ainda mais o processo(ou seja, reduzir ainda mais a variabilidade) é testada, e uma amostra aleatória simples de 24 latas resulta nas quantidades listadas abaixo.

### a. Especifique as hipóteses nula e alternativa apropriadas e faça o teste 
de hipótes
#### Hipótese Nula (H₀): O desvio padrão da nova máquina é igual ao desvio padrão da máquina antiga (σ >= 0,051).
#### Hipótese Alternativa (H₁): O desvio padrão da nova máquina é menor que o desvio padrão da máquina antiga (σ < 0,051).e

### b. Faça as validações necessárias para utilizar o teste.

In [22]:
amostra = [11.98, 11.98, 11.99, 11.98, 11.90, 12.02, 11.99, 11.93,
           12.02, 12.02, 12.02, 11.98, 12.01, 12.00, 11.99, 11.95,
           11.95, 11.96, 11.96, 12.02, 11.99, 12.07, 11.93, 12.05]

n = len(amostra)
variancia_amostral = np.var(amostra, ddof=1) 
desvio_padrao_antigo = 0.051
variancia_antiga = desvio_padrao_antigo**2

chi2_statistic = (n - 1) * variancia_amostral / variancia_antiga

p_valor = stats.chi2.cdf(chi2_statistic, df=n - 1)

print(f"Estatística Qui-Quadrado: {chi2_statistic:.4f}")
print(f"p-valor: {p_valor:.4f}")

Estatística Qui-Quadrado: 13.6470
p-valor: 0.0635


### c. Com um nível de significância de 5%, qual é a conclusão do teste?
#### Como o p-valor (0,0635) é muito maior que o nível de significância (0,05), não rejeitamos a hipótese nula.

## 6. Entre 2200 passageiros de carro do sexo masculino com a idade acima de 8 anos, 72% usam cinto de segurança, entre 2380 passageiros de carro do sexo feminino com idade acima de 8 anos, 84% usam cinto de segurança. Use o nível de significância de 5%, para testar a afirmativa que ambos os sexos têm a mesma taxa no uso de cinto de segurança. Com base no resultado parece haver uma diferença entre os sexos? Não se esqueça de escrever as hipóteses do teste.

In [30]:
n1 = 2200
x1 = 0.72 * n1
p1_hat = 0.72

n2 = 2380
x2 = 0.84 * n2
p2_hat = 0.84

p_hat = (x1 + x2) / (n1 + n2)

z = (p1_hat - p2_hat) / np.sqrt(p_hat * (1 - p_hat) * (1/n1 + 1/n2))

p_valor = 2 * (1 - stats.norm.cdf(abs(z)))

alpha = 0.05

resultado = "Rejeita a hipótese nula" if p_valor < alpha else "Não rejeita a hipótese nula"

print(f"Estatística z: {z:.4f}")
print(f"p-valor: {p_valor:.4f}")
print(resultado)

Estatística z: -9.8327
p-valor: 0.0000
Rejeita a hipótese nula


## 7. Uma distribuidora de combustível deseja verificar se um novo tipo de gasolina 
é eficaz na revitalização de motores velhos. Com esse objetivo, seleciona 1 
automóveis de um mesmo modelo com mais de 8 anos de uso, e ap s
regular os motores, verifica o consumo de combustível. Em seguida, os
carros são abastecidos com o novo tipo de combustível durante 15 sema as,
e uma nova aferição do consumo é feita. Considere as medições da t bela
abaixo, escreva as hipóteses e verifique através de um teste de hipótes  se o
rendimento médio é de fato superior para esse novo combustível ao n vel de
5% de signifcância. 

In [32]:
consumo_antes = np.array([8.1, 7.9, 6.8, 7.8, 7.6, 7.9, 5.7, 8.4, 8.0, 9.5, 8.0, 6.8])
consumo_depois = np.array([11.6, 8.8, 9.9, 9.5, 11.6, 9.1, 10.6, 10.8, 13.4, 10.6, 10.5, 11.4])

diferencas = consumo_depois - consumo_antes

media_diferencas = np.mean(diferencas)
desvio_padrao_diferencas = np.std(diferencas, ddof=1) 
n = len(diferencas)

shapiro_test = stats.shapiro(diferencas)
print(f"Teste de Shapiro-Wilk: Estatística = {shapiro_test.statistic:.4f}, p-valor = {shapiro_test.pvalue:.4f}")

t_statistic = media_diferencas / (desvio_padrao_diferencas / np.sqrt(n))

p_valor = 1 - stats.t.cdf(t_statistic, df=n - 1)

alpha = 0.05

resultado = "Rejeita a hipótese nula" if p_valor < alpha else "Não rejeita a hipótese nula"

print(f"Média das diferenças: {media_diferencas:.4f}")
print(f"Desvio padrão das diferenças: {desvio_padrao_diferencas:.4f}")
print(f"Estatística t: {t_statistic:.4f}")
print(f"p-valor: {p_valor:.4f}")
print(resultado)

Teste de Shapiro-Wilk: Estatística = 0.9397, p-valor = 0.4946
Média das diferenças: 2.9417
Desvio padrão das diferenças: 1.5582
Estatística t: 6.5396
p-valor: 0.0000
Rejeita a hipótese nula


## 8. Digitadores são treinados em uma empresa em duas turmas distintas. Na primeira, denominada turma J, utiliza-se o método japonês de ensino, na segunda turma, denominada turma A, utiliza-se um método alemão. Deseja-se comparar os dois métodos e para tanto, 16 alunos de cada turma foram escolhidos aleatoriamente e uma mesma tarefa foi atribuída a cada um. Ao final do experimento, o tempo gasto na realização da tarefa, para cada aluno foi anotado. No processo, dois computadores utilizados por alunos da turma J e três da turma A, apresentaram problemas que impediram a realização da tarefa, por isso o tamanho da amostra foi reduzido. Apesar de não conhecidas, as variâncias populacionais para as duas turmas são consideradas iguais com base em estudos anteriores. Ao nível de 5%, teste a hipótese para checar se existe diferença significativa no tempo médio de realização da tarefa para as duas turmas. (Sabe-se, que o tempo médio para realização das tarefas tem distribuição normal.) Os dados obtidos foram

In [39]:
import numpy as np
from scipy import stats

turma_j = [10, 13, 9, 10, 14, 13, 10, 15, 12, 10, 9, 10, 13, 14]
turma_a = [15, 12, 18, 16, 15, 17, 17, 15, 16, 17, 11, 17, 14]

n_j = len(turma_j)
n_a = len(turma_a)

media_j = np.mean(turma_j)
media_a = np.mean(turma_a)

var_j = np.var(turma_j, ddof=1) 
var_a = np.var(turma_a, ddof=1) 

sp2 = ((n_j - 1) * var_j + (n_a - 1) * var_a) / (n_j + n_a - 2)
t = (media_j - media_a) / np.sqrt(sp2 * (1/n_j + 1/n_a))

p_valor = 2 * (1 - stats.t.cdf(abs(t), df=n_j + n_a - 2))

alpha = 0.05

resultado = "Rejeita a hipótese nula" if p_valor < alpha else "Não rejeita a hipótese nula"

print(f"Estatística t: {t:.4f}")
print(f"p-valor: {p_valor:.4f}")
print(resultado)

Estatística t: -4.7965
p-valor: 0.0001
Rejeita a hipótese nula


## 9. Queremos verificar o efeito do tipo de impermeabilização em lajes de 
concreto. As quantidades de água que passaram pela laje, em cada tipo d 
concreto, foram medidas obtendo-se os valores da tabela abaix Teste ao nível de 5% de significância a impermeabilidade média dos tipos de 
concreto. Escreva as hipóteses do teste e valide as suposições necessárias 
Qual a conclusão podemos obter?o.

In [45]:
tipo_i = np.array([56, 55, 62, 59, 60])
tipo_ii = np.array([64, 61, 50, 55, 56])
tipo_iii = np.array([45, 46, 45, 39, 43])
tipo_iv = np.array([42, 39, 45, 43, 41])

f_statistic, p_valor = stats.f_oneway(tipo_i, tipo_ii, tipo_iii, tipo_iv)

alpha = 0.05

resultado = "Rejeita a hipótese nula" if p_valor < alpha else "Não rejeita a hipótese nula"

print(f"\nEstatística F: {f_statistic}")
print(f"p-valor: {p_valor}")
print(resultado)


Estatística F: 29.79002624671912
p-valor: 8.775405934817607e-07
Rejeita a hipótese nula


## 10. Um órgão educacional do governo decidiu comparar o desempenho de notas 
obtidas por estudantes de escola particular e pública, a suposição é de qu 
as notas de escola pública são menores que as notas obtidas por estudant s
de escolas particulares. Para testar essa hipótese, foram selecionadas d as
mostras de estudantes que prestaram vestibular, suas médias gerais f ram
categorizadas em 4 grupos e portanto obteve-se os dados a. Teste a hipótese que as duas populações são homogêneas, para o nível de 
significância de 1%.baixo

In [48]:
observado = np.array([[15, 22, 18, 3], [6, 10, 20, 6]])
esperado = np.array([[12.18, 18.56, 22.04, 5.22], [8.82, 13.44, 15.96, 3.78]])

chi2_statistic = np.sum((observado - esperado)**2 / esperado)

graus_liberdade = (observado.shape[0] - 1) * (observado.shape[1] - 1)

p_valor = 1 - stats.chi2.cdf(chi2_statistic, graus_liberdade)

alpha = 0.01

resultado = "Rejeita a hipótese nula" if p_valor < alpha else "Não rejeita a hipótese nula"

print(f"Estatística Qui-Quadrado: {chi2_statistic}")
print(f"Graus de liberdade: {graus_liberdade}")
print(f"p-valor: {p_valor}")
print(resultado)

Estatística Qui-Quadrado: 7.0837500154326705
Graus de liberdade: 3
p-valor: 0.06927574363872058
Não rejeita a hipótese nula
